components/brave_wallet/browser/zcash/zcash_keyring.cc

```diff
@@ -84,4 +184,28 @@ std::optional<std::vector<uint8_t>> ZCashKeyring::SignMessage(
   return hd_key->SignDer(message);
 }
 
+void ZCashKeyring::ConstructRootHDKey(const std::vector<uint8_t>& seed,
+                                      const std::string& hd_path) {
+  HDKeyring::ConstructRootHDKey(seed, hd_path);
+
+#if BUILDFLAG(ENABLE_ORCHARD)
+  if (!seed.empty() && IsZCashShieldedEnabled()) {
+    auto orchard_key = HDKeyZip32::GenerateFromSeed(seed);
+    if (!orchard_key) {
+      NOTREACHED();
+      return;
+    }
+    orchard_key = orchard_key->DeriveHardenedChild(kZip32Purpose);
+    if (!orchard_key) {
+      NOTREACHED();
+      return;
+    }
+    orchard_key_ = orchard_key->DeriveHardenedChild(
+        testnet_ ? kTestnetCoinType
+                 : static_cast<uint32_t>(mojom::CoinType::ZEC));
+    DCHECK(orchard_key_);
```

Comment:
i assume if we are sure this is non-null, we'd better follow as above i.e. just CHECK